src/Utilities/SafeMethods.php

```diff
@@ -26,15 +36,27 @@ public function json_decode(
         int $depth = self::JSON_DECODE_DEPTH,
         int $options = self::JSON_DECODE_OPTIONS
     ) {
-        return \Safe\json_decode($json, $assoc, $depth, $options);
+        $data = \json_decode($json, $assoc, $depth, $options);
+        if (JSON_ERROR_NONE !== json_last_error()) {
```

Comment:
I'd try to see if you can get this `json_last_error()` logic to be inside of `JsonException` so you can encapsulate all that logic there, now it's spread across two places. Alternatively you could think about doing something like:

```
throw new JsonException(json_last_error_msg(), json_last_error()); 
```

here instead, to still keep it in 1 place